tempesta_fw/cache.c

```diff
@@ -190,6 +191,58 @@ tfw_cache_msg_cacheable(TfwHttpReq *req)
 	return __cache_method_test(req->method);
 }
 
+/*
+ * Match the given request against strings specified in "location",
+ * "cache_bypass", and "cache_fulfil" directives and according to
+ * match operators specified in the directives.
+ *
+ * Return the constant that identifies the action on a request.
+ * Currently the actions are "bypass", "fulfill", or "default".
+ */
+tfw_stmt_t
+tfw_cache_action(TfwHttpReq *req)
+{
+	TfwCfgLocation *loc;
+	TfwCfgCacheMatch *cam;
+
+	loc = tfw_location_match(&req->uri_path);
+	cam = tfw_camatch_match(loc, &req->uri_path);
+	if (cam)
+		return cam->stmt;
+	return TFW_D_CACHE_DEFAULT;
+}
+
+/*
+ * Decide if the cache can be employed. For a request that means
+ * that it can be served from cache if there's a cached response.
+ * For a response it means that the response can be stored in cache.
+ *
+ * Various cache action/control directives are consulted when making
+ * the resulting decision.
+ */
+static bool
+tfw_cache_employ(TfwHttpReq *req, TfwHttpResp *resp)
+{
+	/*
+	 * Process requests according to "location", "cache_bypass",
+	 * and "cache_fulfill" directives.
+	 */
+	if (!resp) {
+		tfw_stmt_t stmt = tfw_cache_action(req);
+		if ((stmt == TFW_D_CACHE_DEFAULT)
+		    || (stmt == TFW_D_CACHE_BYPASS))
+		{
+			req->cache_ctl.flags |= TFW_HTTP_CC_NO_STORE;
```

Comment:
Sure. The flag is set by RFC, while the functionality isn't about RFC. So we should not mangle the flag in this patch.